### PR TITLE
Upgrade Travis build to Ubuntu Xenial and Python 3.6

### PR DESCRIPTION
**Pull request by gidsg**
### Context
https://trello.com/c/OfChhRGO/1628-update-orj-build-to-python-36

### Changes proposed in this pull request
* Use Ubuntu Xenial
* Use Python 3.6 native virtualenv
* Test fixes

### Guidance to review
* Build should pass in Travis CI

### CODE REVIEW

File: src/test/java/uk/gov/register/store/postgres/PostgresDataAccessLayerTest.java [hunk 1]
```diff
@@ -40,12 +41,13 @@
     private ObjectMapper objectMapper = new ObjectMapper();
 
     @Rule
-    public WipeDatabaseRule wipeDatabaseRule = new WipeDatabaseRule(TestRegister.address);
+    public RegisterRule register = new RegisterRule();
```
Comment (gidsg) [inline src/test/java/uk/gov/register/store/postgres/PostgresDataAccessLayerTest.java]: this was a similar issue to that in https://github.com/openregister/openregister-java/pull/564#discussion_r244588862 whereby the test attempted to manipulate database state which not exist depending on what order tests are run.

File: .travis.yml [hunk 2]
```diff
@@ -8,19 +7,14 @@ services:
...
-    - python3.4-venv
 cache:
   directories:
   - $HOME/.gradle/caches
   - $HOME/.gradle/wrapper
   - $HOME/.m2
 before_script:
 # work around https://github.com/travis-ci/travis-ci/issues/8363
-- pyenv global system 3.5
+- pyenv global system 3.6
```
Comment (arnau) [inline .travis.yml]: Why is this needed if you use python3's venv?

Comment (gidsg) [reply .travis.yml]: `pyenv` sets the python version used, as there are multiple versions installed on the Travis CI image and 3.6 is not the default. Please see also discussion in https://github.com/travis-ci/travis-ci/issues/8363
